### PR TITLE
Additional Device Kernels

### PR DESCRIPTION
**Pull request by tomstitt**
- adds `MultAtB` and `MultT` (vector)
   - happy to change the names, especially for `MultT`
   - we have these locally but I think they make more sense in mfem itself
- adds `const` qualifier to the `data` param in `Mult` (vector)

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2423](https://github.com/mfem/mfem/pull/2423) | @tomstitt | @tzanio | @camierjs + @v-dobrev + @YohannDudouit | 07/29/21 | 10/31/21 | 11/02/21 | |

### CODE REVIEW

**Review by artv3 — APPROVED** [review]

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

File: linalg/kernels.hpp [hunk 1]
```diff
@@ -188,6 +188,35 @@ void Mult(const int height, const int width, TA *data, const TX *x, TY *y)
    }
 }
 
+/** @brief Matrix transpose vector multiplication: y = At x, where the matrix A
+    is of size @a height x @a width with given @a data, while @a x and @a y
+    specify the data of the input and output vectors. */
+template<typename TA, typename TX, typename TY>
+MFEM_HOST_DEVICE inline
+void MultT(const int height, const int width, const TA *data, const TX *x,
```
Comment (YohannDudouit) [inline linalg/kernels.hpp:196]: `MultTranspose` seems to be more idiomatic in the MFEM code.

File: linalg/kernels.hpp [hunk 2]
```diff
@@ -188,6 +188,35 @@ void Mult(const int height, const int width, TA *data, const TX *x, TY *y)
...
+           TY *y)
+{
+   if (width == 0)
+   {
+      for (int row = 0; row < height; row++)
+      {
+         y[row] = 0.0;
+      }
+      return;
+   }
```
Comment (YohannDudouit) [inline linalg/kernels.hpp:206]: The matrix is transposed so I think `height` and `width` should be inverted:
```suggestion
   if (height == 0)
   {
      for (int row = 0; row < width; row++)
      {
         y[row] = 0.0;
      }
      return;
   }
```

**Review by YohannDudouit — CHANGES_REQUESTED** [review]

**Comment by tzanio** [comment]
ping: @tomstitt

File: linalg/kernels.hpp (hunk 2 above)
Comment (tomstitt) [reply linalg/kernels.hpp:206]: yes!! thank you! i should have been more careful when copy/pasting this part=p

**Comment by tzanio** [comment]
@YohannDudouit and @v-dobrev -- can you please review?

**Review by camierjs — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Review by tzanio — APPROVED** [review]

**Pull request merged**